### PR TITLE
Use succinct data structure decoder as default.

### PR DESCRIPTION
**Pull request by mankyKitty**
The succinct data structure work is so much faster than the traversal based
decoder that it seems wild to not have it as the default.

Begin merging the succinct data structure decoder back in and start to fix
things up.

### CODE REVIEW

File: waargonaut.cabal [hunk 1]
```diff
@@ -137,15 +144,17 @@ test-suite doctests
...
 
-  build-depends:       base >= 4.7 && < 4.13
-                     , waargonaut
+  build-depends:       base             >= 4.7 && < 4.13
                      , doctest          >= 0.9.7
                      , filepath         >= 1.3
                      , directory        >= 1.1
                      , template-haskell >= 2.8
-                     , parsec           >= 3.1
+                     , hedgehog         >= 0.6    && < 0.7
```
Comment (gwils) [inline waargonaut.cabal]: The spacing here doesn't line up with the others

File: waargonaut.cabal [hunk 2]
```diff
@@ -119,12 +121,17 @@ library
...
                      , transformers
                      , bifunctors
                      , contravariant
                      , wl-pprint-annotated
 
+                     , hw-json >= 0.6
+                     , hw-prim
+                     , hw-balancedparens
+                     , hw-rankselect
+                     , hw-bits
```
Comment (gwils) [inline waargonaut.cabal]: It is definitely worth putting bounds on these before release, but for now it should be ok.

File: waargonaut.cabal [hunk 3]
```diff
@@ -190,19 +199,24 @@ test-suite waarg-tests
...
+  hs-source-dirs:      test
   main-is:             Bench.hs
 
   build-depends:       base         >= 4.7   && < 4.13
                      , lens         >= 4.15  && < 5
-                     , parsec       >= 3.1
                      , semigroups   >= 0.8.4 && < 0.19
                      , text         >= 1.2   && < 1.3
+                     , hedgehog     >= 0.6    && < 0.7
+                     , generics-sop >= 0.3.2  && < 4
```
Comment (gwils) [inline waargonaut.cabal]: I think you mean `< 0.4`

Comment (mankyKitty) [reply waargonaut.cabal]: No I didn't!!

**NARRATOR**: He did.

**Pull request merged**